AutocompleteClient/FW/Logic/Request/CIOTrackQuizResultClickData.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  CIOTrackQuizResultClickData.swift
+//  Constructor.io
+//
+//  Copyright © Constructor.io. All rights reserved.
+//  http://constructor.io/
+//
+
+import Foundation
+
+/**
+ Struct encapsulating the parameters that must/can be set set in order to track quiz result click
+ */
+struct CIOTrackQuizResultClickData: CIORequestData {
+
+    let quizID: String
+    let quizVersionID: String
+    let quizSessionID: String
+    let customerID: String
+    let variationID: String?
+    let itemName: String?
+    let resultID: String?
+    let resultPage: Int?
+    let resultCount: Int?
+    let numResultsPerPage: Int?
+    let resultPositionOnPage: Int?
+    var sectionName: String?
+
+    func url(with baseURL: String) -> String {
+        return String(format: Constants.trackQuizResultClick.format, baseURL)
+    }
+
+    init(quizID: String, quizVersionID: String, quizSessionID: String, customerID: String, variationID: String? = nil, itemName: String? = nil, resultID: String? = nil, resultPage: Int? = nil, resultCount: Int? = nil, numResultsPerPage: Int? = nil, resultPositionOnPage: Int? = nil, sectionName: String? = nil) {
+        self.quizID = quizID
+        self.quizVersionID = quizVersionID
+        self.quizSessionID = quizSessionID
+        self.customerID = customerID
+        self.variationID = variationID
+        self.itemName = itemName
+        self.resultID = resultID
+        self.resultPage = resultPage
+        self.resultCount = resultCount
+        self.numResultsPerPage = numResultsPerPage
+        self.resultPositionOnPage = resultPositionOnPage
+        self.sectionName = sectionName
+    }
+
+    func decorateRequest(requestBuilder: RequestBuilder) {
+        requestBuilder.set(autocompleteSection: self.sectionName)
+    }
+
+    func httpMethod() -> String {
+        return "POST"
+    }
+
+    func httpBody(baseParams: [String: Any]) -> Data? {
+        var dict = [
+            "quiz_id": self.quizID,
+            "quiz_version_id": self.quizVersionID,
+            "quiz_session_id": self.quizSessionID,
+            "item_id": self.customerID,
+            "action_class": "result_click"
```

Comment:
Let's remove `action_class` here since it defaults to it on the backend. No reason for us to specify it here as well.